sphinx/domains/javascript.py

```diff
@@ -106,20 +106,19 @@ def add_target_and_index(self, name_obj: Tuple[str, str], sig: str,
                              signode: desc_signature) -> None:
         mod_name = self.env.ref_context.get('js:module')
         fullname = (mod_name + '.' if mod_name else '') + name_obj[0]
-        if fullname not in self.state.document.ids:
-            signode['names'].append(fullname)
```

Comment:
This PR also removes hypertarget names for JS objects. I don't know this is official feature or not. But it works well at this moment.
```
.. js:function:: func

   blah blah blah

reference with func_.
```

But I don't consider this is not intended behavior. Users should use `:js:*:` role instead to refer them.
So it is disabled in this PR.

Reply:
I noticed we have to mention about this on CHANGES.

Reply:
updated.

Reply:
>this is not intended behavior

Agreed